Data:
The manufacturer tests the durability of 15 print-heads. The data (durability in millions of characters before failure) is:

1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

Task a: Build 99% Confidence Interval Using Sample Standard Deviation

Steps:
Compute Sample Mean 
Compute Sample Standard Deviation (s)
t-Distribution
Calcualting Confidence Interval 

In [1]:
import numpy as np
import scipy.stats as stats

# Data
durability = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

# Sample size
n = len(durability)

# Sample mean and sample standard deviation
mean = np.mean(durability)
std_dev = np.std(durability, ddof=1)

# t critical value for 99% confidence and n-1 degrees of freedom
t_critical = stats.t.ppf(0.995, df=n-1)

# Margin of error
margin_of_error = t_critical * (std_dev / np.sqrt(n))

# Confidence Interval
ci_lower = mean - margin_of_error
ci_upper = mean + margin_of_error

mean, std_dev

(1.2386666666666666,
 0.19316412956959936,
 2.97684273411266,
 (1.090197338451367, 1.3871359948819662))

In [3]:
t_critical, (ci_lower, ci_upper)

(2.97684273411266, (1.090197338451367, 1.3871359948819662))

Since the sample size is small (15) and the population standard deviation is unknown, the t-distribution is appropriate.
The t-distribution accounts for the added uncertainty that comes from estimating the population standard deviation from the sample

Task b: Build 99% Confidence Interval Using Known Population Standard Deviation

Steps: Known population standard deviation ,z-Distribution,Calculating the Confidence Interval

In [2]:
# Known population standard deviation
pop_std_dev = 0.2

# z critical value for 99% confidence
z_critical = stats.norm.ppf(0.995)

# Margin of error
margin_of_error_known_std = z_critical * (pop_std_dev / np.sqrt(n))

# Confidence Interval
ci_lower_known_std = mean - margin_of_error_known_std
ci_upper_known_std = mean + margin_of_error_known_std

z_critical, (ci_lower_known_std, ci_upper_known_std)

(2.5758293035489004, (1.1056514133957607, 1.3716819199375725))

Results:

In [4]:
t_critical, (ci_lower, ci_upper)

(2.97684273411266, (1.090197338451367, 1.3871359948819662))

99% Confidence Interval ((1.090197338451367, 1.3871359948819662)) . This means that we are 99% confident that the true mean durability of the print-heads lies between 1.090 and 1.387 million characters.

In [5]:
z_critical, (ci_lower_known_std, ci_upper_known_std)

(2.5758293035489004, (1.1056514133957607, 1.3716819199375725))

99% Confidence Interval:(1.106,1.372)
This means that, assuming the population standard deviation is known, we are 99% confident that the true mean durability lies between 1.106 and 1.372 million characters.

In both cases, the confidence intervals provide a range of values where the true mean durability of the print-heads is likely to fall